src/getINP.ts

```diff
@@ -14,118 +14,127 @@
 
-  const entryHandler = (entry: PerformanceEventTiming) => {
-    // TODO: Perhaps ignore values before FCP
-    if (!entry.interactionId) return;
-    
-    updateInteractionIds(entry.interactionId);
-    addInteractionEntryToINPList(entry);
-
-    const inpEntry = getCurrentINPEntry();
-
-    // Only report when the IMP value changes.  However:
-    // * When we cross a %-ile boundary, pushing `which` up, or
-    // * A new long value is added to the top, moving the current INP entry down
-    // ...then the inpEntry will change, but `duration` value of the new entry may still be the same.
-    // While technically the INP metric.value doesn't change, we still report since metric.entries changes.
-    //
-    // Potentially, we may even want to compare the whole metric.entries range for equality, because:
-    // * We can have cases where a middle value updates due to new-longest value with same interactionId.
-    // * When we are already at MAX_ENTRIES and `which` stops changing, but the current smallest can get popped off.
-    const which = largestINPEntries.indexOf(inpEntry);
-    if (which >= metric.entries.length || metric.value != inpEntry.duration) {
-      metric.value = inpEntry.duration;
-      // We attach all the longest responsiveness entries, not just the HighP value.
-      // While technically the INP score is exactly the entry.duration of one specific HighP-ile entry...
-      // the entry would not have been picked (and IMP would be lower) if *any* of the worst entries were not so high.
-      // Improving any of them will improve score.
-      metric.entries.length = 0;
-      metric.entries.push(...largestINPEntries.slice(0, which + 1));
-    }
+  const handleEntries = (entries: Metric['entries']) => {
+    (entries as PerformanceEventTiming[]).forEach((entry) => {
+      if (entry.interactionId) {
+        processEntry(entry);
+      }
+    });
 
-    // Perhaps Event Timing is the first API that can have multiple entries in a single PO callback
-    // That means that we would ideally report() only after the whole list of entries is processed, not one per entry.
-    // If we were lucky, the entries would be in timestamp order so the first is the longest... but I've found they
-    // are ordered in other ways... by type, I think?
-    // Alternatively: sort entries in the observe() wrapper.
-    report();
+    const inp = estimateP98LongestInteraction();
+
+    if (inp && inp.latency !== metric.value) {
+      metric.value = inp.latency;
+      metric.entries = inp.entries;
+      report();
+    }
   };
 
-  const po = observe('event', entryHandler as PerformanceEntryHandler);
+  const po = observe('event', handleEntries, {
+    // The use of 50 here as a balance between not wanting the callback to
+    // run for too many already fast events (one frame or less), but also
+    // get enough fidelity below the recommended "good" threshold of 200.
+    durationThreshold: 50,
```

Comment:
Event Timing is rounded to the nearest 8ms.  That makes the duration a nice round number (either 48ms or 56ms). 
 That means that this 50ms durationThreshold is effectively a 56ms duration threshold, which we may want to be explicit about.

TL;DR I think this is a good threshold.  But here are some nuances:

* At 60hz, 56ms is 3.5 frames, which means it can effectively get presented at the same time as other entries with 64ms of latency.
* Conversely, if you were to change to 48ms durationThreshold, you really should probably use 40ms to include entries that take 2.5 frames @ 60hz.
* If you wanted to group entries by frame (i.e. by presentation time) you would need to take startTime + duration and then group entries which are within 8ms windows.  A single shared presentation time can include entries with a wide range of durations.  An entry with less than `durationThreshold` duration can have large processing time.  (This is one reason why we may want a dedicated api)

Reply:
Done.